src/Mentat.UI/Controllers/CardController.cs

```diff
@@ -82,38 +79,35 @@ public ActionResult Edit(string id)
         }
 
         // POST: CardController/Edit/5
+        // Edit step 2
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public ActionResult Edit(string id, IFormCollection collection)
+        public ActionResult Edit(string id, [FromForm] Card card)
         {
             try
             {
```

Comment:
This is the source of the bug. You're sending a Card directly from the form to the DB; the FromForm card won't have an ID attached to it (Card card.Id == null).

Before passing it, do:
```
card.Id = id;
```

Reply:
@colewinfield somehow you flipped this from a draft to a PR. Do you know how to switch it back? This will not be a pull-request.

Reply:
I am closing this pull request - please implement the changes to the development branch.

Reply:
@Freytes switching it back to a draft would have been optimal. What do you mean by 'please implement the changes to the development branch'?